testing/regression/gh_902__iterator_category_to_system_returns_wrong_tag.cu

```diff
@@ -0,0 +1,22 @@
+#include <thrust/iterator/detail/iterator_category_with_system_and_traversal.h>
+#include <thrust/iterator/iterator_categories.h>
+#include <thrust/iterator/iterator_traits.h>
+
+template <class IteratorCategory>
+using _category_to_system_t =
+    typename thrust::detail::iterator_category_to_system<IteratorCategory>::type;
+
+static_assert(std::is_same<thrust::device_system_tag, _category_to_system_t<thrust::output_device_iterator_tag>>::value);
```

Comment:
We need to build with C++11 + C++14, so these will need to use the `static_assert(cond, msg)` syntax.

Reply:
Looks like CI didn't catch this because the tests is only added when the `thrust` target is defined in `THRUST_TARGETS`. For multiconfig builds (used by CI), this target won't exist and we'll instead have `thrust.[host].[device].[dialect]` targets.